### ISSUE TITLE
Strange merge of fulljones h5parms

### ISSUE THREAD

**Issue by tikk3r** [revoltek/losoto#105]
Initiall reported here: https://github.com/lmorabit/lofar-vlbi/issues/70

After converting killMS solutions to fulljones h5parms with killMS2H5parm.py, merging these with H5parm_collector.py behaves a bit strange. For example, in the plots below the merged h5parm shows nothing for 164 MHz at an arbitrary timeslot, yet there are valid solutions there in the specific 164 MHz h5parm.

It appears that it gets confused about the axes or something like that, giving adjacent timeslots different "chunks of frequency". Converting them as only XX and YY with --nofulljones does not show this problem.

The plots show the merged h5parm in the first row and two example h5parms from individual 2 MHz blocks in the second row.

| Time slot 2| Time slot 3|
:-: | :-:
![CS301_HBA1_concatenated_ddf_solutions2](https://user-images.githubusercontent.com/2464769/97581823-59404c00-19f5-11eb-8f1f-3f8521b9e102.png) | ![CS301_HBA1_concatenated_ddf_solutions](https://user-images.githubusercontent.com/2464769/97581825-59d8e280-19f5-11eb-8553-6d5c1c5dfced.png)

150 MHz block timeslot 2 | 164 MHz block timeslot 2
:-: | :-:
![CS301_HBA1_164MHz_ddf_solutions](https://user-images.githubusercontent.com/2464769/97581826-5a717900-19f5-11eb-948b-e35422a02d3d.png) | ![CS301_HBA1_150MHz_ddf_solutions](https://user-images.githubusercontent.com/2464769/97581828-5b0a0f80-19f5-11eb-94ef-ea93a796a1ef.png)

**Comment by revoltek** [comment]
yes, most likely it's a problem of axes ordering. I have tested that code only with LBA and only in a specific format. Are all h5parms at least with the same axes order/polarisations? It's dangerous it doesn't show any error though...

**Comment by tikk3r** [comment]
They have the same structure. I just realize I already mention a possible answer in the other issue. The axes do have the same order, but they do not have the same length necessarily. The kMS solutions can have different time and frequency axes per block. I guess the collector does not like that?

**Comment by revoltek** [comment]
I use the collector only to merge equal tables, have a look at the H5parm_interpolator.py, which has also a number of limitations. Coding a very flexible tool to merge tables is a bit of a pain, currently I've only these 2 ready

**Comment by tikk3r** [comment]
I tried H5parm_interpolator out, but it didn't quite do what I expected. What is it intended to do? I tried running it and it produced a single h5parm with apparently all the solutions added/multiplied (it seems) into a single solset and single amplitude and phase soltabs. The frequency axis only has the range of the first one and the amplitudes seem to all have been multiplied or something as now they are suddenly of order 100 where they are of order 1 in the input H5parms.

H5parm_interpolator sounds like what I'm looking for though, so I'll poke around in it a bit more.

**Comment by revoltek** [comment]
I use it to take all fast phase and slow amp solutions from many directions and combine them into a single h5parm to be used with DDF - I tested it only in a few cases and there's a number of possible restrictions that I probably never though about... if you come up with a more general version that would be great

**Comment by tikk3r** [comment]
Ah I see, that makes sense.

I think because each h5parm covers a different frequency range it "breaks" the interpolation in this case as well as that makes it not just interpolating to an identical, but faster grid but needing to interpolate different ranges of values to a common interval. I'm definitely in a non-standard messy case I think.

**Comment by tikk3r** [comment]
> [@tikk3r](https://github.com/tikk3r), is this still an issue that needs to be fixed in losoto? Or can this ticket be closed?

I don't see any substantial changes to `H5parm_collector.py` over the past 5 or so years, so the issue is probably still there? In our processing we have switched to a different tool for merging h5parms now, however, so I haven't gone back to this in a very long time.

**Comment by adrabent** [comment]
@tikk3r 
what is the different tool capable of what the standard losoto one is not? Shouldn't we migrate to a common tool?
